### DISCUSSION TITLE
unsafe built-in function calls in expression: http.send

### DISCUSSION THREAD

**Discussion by JinXiu2023** [open-policy-agent/community#431] [OPA and Rego]
It caused the error "unsafe built-in function calls in expression: http.send" when I used the function `http.send()`. There is only one command and I don't know where the error is .

I can reproduce the exception by the following code.
``` go
package example

response := http.send({"method": "get", "url": "http://xxx.com"})
```
And the output is :
``` go
Error Evaluating.
1 error occurred: policy.rego:4: rego_type_error: unsafe built-in function calls in expression: http.send
```
Thank you very much.

**Comment by srenatus** [comment]
Try this instead,

```rego
package example

response := res {
    res := http.send({"method": "get", "url": "http://xxx.com"})
}
```

I suspect it's a bug related to how the parser turns naked expressions into rules. If this ☝️ works, please file an issue -- your original example _should_ work, too.

**Reply by anderseknert** [reply]
I'm not able to reproduce this. @JinXiu2023 what command are you running to get this? 

"Error Evaluating." does not look like something OPA would output. Is this Gatekeeper or some other app embedding OPA? Some of them disable built-ins like `http.send`.

**Reply by JinXiu2023** [reply]
@srenatus Thank you, but this doesn't work either. 
@anderseknert I use [The Rego Playground](https://play.openpolicyagent.org/#L11) to evaluate the policy. You could have a try to reproduce the error.
![image](https://github.com/open-policy-agent/community/assets/132969481/e296ca86-6064-4b59-9694-8812511cae17)

**Reply by anderseknert** [reply]
Yep, `http.send` is not supported on the Rego Playground for security reasons. If you run your policy using `opa eval` it should work just fine.

**Reply by JinXiu2023** [reply]
Ok. Thank you and I will have a try.
